File: BACKEND/app/Models/Transaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Transaction extends Model
{
    protected $fillable = [
        'user_id',
        'title',
        'amount',
        'type',
        'category',
        'date',
        'note',
    ];

    protected $casts = [
        'title' => 'encrypted',
        'note' => 'encrypted',
        'amount' => 'float',
        'date' => 'integer',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: FRONTEND/app/src/main/java/com/myscratch/app/domain/model/Transaction.kt
package com.myscratch.app.domain.model

data class Transaction(
    val id: String,
    val userId: String,
    val title: String,
    val amount: Double,
    val type: TransactionType,
    val category: String,
    val date: Long = System.currentTimeMillis(),
    val note: String = ""
)
